src/praisonai-agents/tests/test_handoff_compatibility.py

```
@@ -0,0 +1,238 @@
+        {"role": "user", "content": "Hello"},
+        {"role": "assistant", "content": "Hi", "tool_calls": [{"name": "test"}]},
+        {"role": "tool", "content": "result"},
+        {"role": "user", "content": "Thanks"}
+    ]
+    
+    # Test remove_all_tools filter
+    data = HandoffInputData(messages=list(messages))
+    filtered = handoff_filters.remove_all_tools(data)
+    assert len(filtered.messages) == 3
+    assert all(msg.get("role") not in ["tool"] and "tool_calls" not in msg 
+              for msg in filtered.messages if isinstance(msg, dict))
```

Flake8 (7.2.0)

[error] continuation line under-indented for visual indent

(E128)